dashboard/src/actions/index.js

```diff
@@ -137,19 +139,18 @@ export const runQuery = query => {
                             // We display the response from server.
                         } else {
                             dispatch(addQuery(query));
-                            dispatch(
-                                saveSuccessResponse(JSON.stringify(result))
-                            );
+                            dispatch(saveSuccessResponse(null, result, true));
```

Comment:
Passing empty string might be more readable instead of explicitly passing `null`